lib/jobs/redfish-job.js

```diff
@@ -0,0 +1,226 @@
+        var self = this;
+        return waterline.workitems.update({name: "Pollers.Redfish"}, {failureCount: 0})
+        .then(function() {
+            self._subscribeRedfishCommand(self.routingKey, function(data) {
+                if (self.concurrentRequests(data.node, data.workItemId)) {
+                    return;
+                }
+                self.addConcurrentRequest(data.node, data.workItemId);
+                assert.object(data.config, 
+                    'Redfish Poller Data Config');
+
+                return waterline.nodes.needByIdentifier(data.node)
+                .then(function(node) {
+                    var obmSetting = _.find(node.obmSettings, { service: 'redfish-obm-service' });
+                    if (obmSetting) {
+                        return obmSetting.config;
+                    }
+                })
+                .then(self.initClient)
+                .then(function(client) {
+                    return self.collectChassisData(
+                        client, 
+                        data.config.command
+                    );
+                })
+                .then(function(result) {
+                    assert.object(result,
+                        'Expected Result Object For: ' + data.config.command
+                    );
+                    data[data.config.command] = result;
+                    return self._publishRedfishCommandResult(
+                        self.routingKey, 
+                        data.config.command, 
+                        data
+                    );
+                })
+                .then(function() {
+                    return waterline.workitems.findOne({ id: data.workItemId });
+                })
+                .then(function(workitem) {
+                        return waterline.workitems.setSucceeded(null, workitem);
+                })
+                .catch(function (err) {
+                    logger.error("Error Retrieving Redfish Data.", {
+                        data: data,
+                        error: err
+                    });
+                })
+                .finally(function() {
+                    self.removeConcurrentRequest(data.node, data.workItemId);
+                });
+            });
+        })
+        .catch(function(err) {
+            logger.error("Failed to initialize job", { 
+                error:err
+            });
+            self._done(err);
+        });
+    };
+
+    /**
+     * @function concurrentRequests
+     * @description manage concurrent work item command requests
+     */
+    RedfishJob.prototype.concurrentRequests = function(node, workItemId) {
+        assert.string(node);
+        assert.string(workItemId);
+        if(!_.has(this.concurrent, node)){
+            this.concurrent[node] = {};
+        }
+        if(!_.has(this.concurrent[node], workItemId)){
+            this.concurrent[node][workItemId] = 0;
+        }
+        if(this.concurrent[node][workItemId] >= this.maxConcurrent){
```

Comment:
Just a minor suggestion, just return this expression is enough, we can save following 4 lines.